plugins/device/rest.go

```diff
@@ -41,6 +42,18 @@ func deviceAddHandler(w http.ResponseWriter, r *http.Request) {
 	}
 	defer txn.Done()
 
+	_, err = deviceutils.GetDevice(peerID, req.Device)
+	if err == nil {
+		logger.WithError(err).WithField("device", req.Device).Error("Device already exists")
+		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, "device already exists")
+		return
+	}
+
+	if err != errors.ErrDeviceNotFound {
+		restutils.SendHTTPError(ctx, w, http.StatusNotFound, err)
```

Comment:
this also could be an internal server error right?

Reply:
ok